internal/controller/clusterpolicy_controller.go

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/edgedb/edgedb-go"
+	"github.com/pingcap/errors"
+
+	kyvernoV1 "github.com/kyverno/kyverno/api/kyverno/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	edgedbutils "github.com/giantswarm/policy-meta-operator/internal/utils/edgedb"
+)
+
+// PolicyReconciler reconciles a AutomatedException object
+type ClusterPolicyReconciler struct {
+	client.Client
+	EdgeDBClient *edgedb.Client
+	Scheme       *runtime.Scheme
+}
+
+func (r *ClusterPolicyReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	_ = log.FromContext(ctx)
+
+	var clusterPolicy kyvernoV1.ClusterPolicy
+
+	if err := r.Get(ctx, req.NamespacedName, &clusterPolicy); err != nil {
+		if !errors.IsNotFound(err) {
+			// Error fetching the report
+			log.Log.Error(err, "unable to fetch Policy")
+		} else {
+			// Policy not found, make sure we don't have it in edgedb either
+			err := edgedbutils.DeletePolicy(ctx, r.EdgeDBClient, req.Name)
```

Comment:
Should be a new Delete() for Kyverno CPols, right?

Reply:
Yeah, I also want to handle when the policy is set back to `Audit` so we delete it too.

Reply:
We have a Delete method now, but now I ask:
Is it worth having ALL policies in edgedb? Right now I'm just storing the ones in Enforce mode. But maybe we have a use case for having all of them there.

Reply:
We'll eventually have all policies in edgedb, for sure. But it might not be polmo putting them there, so could happen now or later